addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2543,6 +2543,11 @@ registry.background = SnippetOptionWidget.extend({
         // TODO should be automatic for all options as equal to the start method
         this.bindBackgroundEvents();
         this.__customImageSrc = this._getSrcFromCssValue();
+        const $bg = this.$target.find('> .s_parallax_bg');
+        if (!this.$target.hasClass('parallax') && $bg.length) {
```

Comment:
Simple efficiency stuff: here you search for the bg, then check if the target has the parallax class then check if you found a bg. Check the bg before the parallax class, so that if there is no bg, the parallax class does not have to be checked.

However, I don't understand why this code is only in one way? If we change the target from the snippet to the bg span, you must transfer the image on the bg and in the other case, you must transfer from the bg to the snippet.

Reply:
We don't need to do it the other way because the switch is done in the parallax animation. When the parallax class is added on the snippet.

Reply:
Explain that in the code ;)